discovery_engine_core/providers/src/bing.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Client to retrieve trending topics.
+
+use serde::{Deserialize, Deserializer, Serialize};
+use url::Url;
+
+use crate::{seal::Seal, Client, Error, Market, Query};
+
+const URL_SUFFIX: &str = "_tt";
+
+/// Parameters for fetching trending news topics.
+pub struct TrendingQuery<'a> {
+    /// Market to fetch results from.
+    pub market: &'a Market,
+}
+
+impl Query for TrendingQuery<'_> {
+    fn setup_url(&self, url: &mut Url) -> Result<(), Error> {
+        url.path_segments_mut()
+            .map_err(|_| Error::InvalidUrlBase(None))?
+            .push(URL_SUFFIX);
+
+        let query = &mut url.query_pairs_mut();
+
+        let country = &self.market.country_code;
+        let lang = &self.market.lang_code;
```

Comment:
:art: since `lang` is used first before `country` in the format string, the order of those two bindings could be switched to logically match it.